sound/soc/sof/sof-priv.h

```diff
@@ -53,6 +53,10 @@ struct snd_sof_pcm_stream;
 							 */
 #define SOF_DBG_DSPLESS_MODE			BIT(15) /* Do not initialize and use the DSP */
 
+#define SOF_DBG_FORCE_LIBRARY_LOAD			BIT(16) /* ignore the DSP D3 persistent capability
+							* and always download library upon D3 exit
+							*/
+
```

Comment:
what's wrong with the `SOF_DBG_IGNORE_D3_PERSISTENT` (bit7)? Yes, it is also going to force re-load the firmware itself.
I would argue that this new flag must not be used in production and it is solely for developers in case the context save and restore (to/from IMR) in firmware is not available.

If we really need a new flag just for libraries then:
```c
#define SOF_DBG_FORCE_LIBRARY_RELOAD			BIT(12)
```
perhaps would be better named and the bit is different (there is an intentional gap before the DSPless mode)

Reply:
@mengdonglin please share more info.    Context save and restore would cost more times and may be not necessary. I am not sure it will be enabled by default or not. Currently it is used for mtl-006 branch since IMR_save is disabled so we need this PR.

Reply:
@aiChaoSONG  @ujfalusi  Actually we always call sof_ipc4_reload_fw_libraries, so we always try to reload library but we don't load for some cases, so I use this SOF_DBG_FORCE_LIBRARY_LOAD, not RELOAD

Reply:
@RanderWang, we only reload libraries in case we did not booted from IMR, yes we call `sof_ipc4_reload_fw_libraries()` unconditionally as the core does not need to know what the platform does and how.
Let's add a flag which is not tied up to library loading, but a generic one indicating that the FW will not retain the context on IMR boot?
The context store/restore is a direct dependency of the libraries after IMR boot, right? So, D3_CONTEXT_LOST or something?

Reply:
sure, good idea! let's use D3_CONTEXT_LOST

Reply:
@RanderWang  are you saying you want to reload library even when booting from IMR? What purpose does that serve when we previously agreed that IMr boot will preserve libraries? What has changed to require this?

Reply:
No, it is for IMR_CONTEXT_SAVE, not IMR_booting.   IMR booting only saves base fw in IMR.  loadable module is loaded into dsp SRAM directly, not in IMR.   IMR_CONTEXT_SAVE save the whole SRAM in IMR including loadable module on D3 entry, and restore it on D3 exit.  So if IMR_CONTEXT_SAVE is disabled, we need to reload library

Reply:
sure, IMR_CONTEXT_SAVE is transparent to kernel. Let's try to sync them.